peekaboo/toolbox/cortex.py

```diff
@@ -56,12 +57,31 @@ def __init__(self, analyzer):
 
 class CortexAnalyzerReport:
     """ Cortex analyzer report base class. """
+
+    report_schema_taxonomies = schema.Schema([{
+        "level": schema.Schema(schema.Or("info", "malicious", "safe")),
+        "namespace": schema.Schema(schema.Or("FileInfo", "VT")),
+        "predicate": str,
+        "value": str
+    }])
+
+    report_schema_artifacts = schema.Schema([{
+        "data": str,
+        "dataType": str,
+    #    "message": schema.Schema(schema.Or(str, None)),
+    #    "tags": list,
+    #    "tlp": int
+    }], ignore_extra_keys=True)
+
     def __init__(self, report):
         if report is None:
             self.report = {}
 
-        if not isinstance(report, dict):
-            raise TypeError('report is expected to be a dict')
+        # we want to make to sure to only provocate a SchemaError
+        try:
+            self.report_schema
+        except AttributeError:
+            self.report_schema = schema.Schema(None)
```

Comment:
Would it be an option here to make `report_schema` a class member variable as well (as it wouldn't change over the lifetime of the report nor be different for individual instances anyway) and have a default for it in the base class that, if not overridden by the child class causes the `validate()` to fail, possibly with a somewhat descriptive error message basically as an indirect `NotImplementedError` so the developer gets a clue what's going on?
``` python
class CortexAnalyzerReport:
    report_schema = schema.Schema(None, error='Subclass needs to provide a schema')
```
would lead to:
```
schema.SchemaError: Subclass needs to provide a schema
```

Reply:
Addressed in recent commit.